test/lib/sstable_test_env.hh

```diff
@@ -185,6 +185,10 @@ public:
             return func(env);
         });
     }
+
+    table_for_tests make_table_for_tests(schema_ptr s = nullptr, std::optional<sstring> dir = std::nullopt) {
```

Comment:
Taking into account my comment above, the dir argument is not used and can be removed. This will discourage users from using explicit tempdir which's good